packages/client/src/util/TypeCasting.ts

```diff
@@ -0,0 +1,65 @@
+import {Field} from 'apache-arrow'
+import {isNumber, isUnsignedNumber} from './common'
+import {Type as ArrowType} from 'apache-arrow/enum'
+
+/**
+ * Function to cast value return base on metadata from InfluxDB.
+ *
+ * @param field the Field object from Arrow
+ * @param value the value to cast
+ * @return the value with the correct type
+ */
+export function getMappedValue(field: Field, value: any): any {
+  if (value === null || value === undefined) {
+    return null
+  }
+
+  const metaType = field.metadata.get('iox::column::type')
+
+  if (!metaType || (!metaType && field.typeId === ArrowType.Timestamp)) {
```

Comment:
I might be mistaken but I think if the first `!metaType` evaluates to true the evaluation of `||` breaks and the second condition containing `!metaType && filed.typeId === ArrowType.Timestamp` is not tested.  Conversely if `!metaType` evaluates to `false` the second statement using `&&` will also be `false`.  

Perhaps, if we want to return early and save unnecessary processing, this could be 
```js
  if (!metaType || field.typeId === ArrowType.Timestamp ) { 
    return value
  }
```